accepted/package-doc/package-doc.md

```diff
@@ -0,0 +1,170 @@
+  might find a new API that is provided as a package and that package might
+  still be in beta.
+
+* **Compatibility should be obvious**. Developers should be able to browse the
+  documentation without having to understand whether a given API is available
+  because it happens to be provided as a .NET Standard-based library or whether
+  it's a framework-specific API. A package that provides APIs to multiple
+  frameworks should therefore show up for all these frameworks, following NuGet
+  compatibility rules.
+
+### Non-Goals
+
+* **Fully unifying frameworks with package-based monikers**. Today, the [API
+  browser] has separate monikers for frameworks (such as .NET Framework and .NET
+  Core) and fully NuGet-based products (such as ASP.NET Core and the Azure SDK).
+  We'll only include NuGet packages in the framework monikers that (1) we
+  consider framework extensions and (2) that aren't big enough to warrant having
+  a moniker on their own.
+
+## Design
+
+We'll provide a drop location for the documentation indexer that will roughly
+look like this:
+
+```
+├───netcore2.0
+│       System.Runtime.dll
+│       System.Collections.dll
+│       ...
+└───extensions
+        System.Collections.Immutable_1.2.3.nukpkg
+        System.Memory_0.1.2-beta.nukpkg
+        ...
+```
+
+The intent is as follows:
+
+* **Frameworks** (.NET Framework, .NET Core, .NET Standard) will be provided as
+  a set of reference assemblies.
+
+* **Framework extensions** are provided as NuGet packages. Each package will
+  have its own version and quality label and the contents of the package will
+  indicate which frameworks the APIs are available for.
+
+We expect the documentation system to index the packages to create a data file
+that maps APIs to frameworks. Logically, the indexer needs to call into NuGet to
+get the list of assemblies for a given framework moniker. If a package doesn't
+support a given framework, that list is considered empty.
+
+The indexing algorithm would roughly look as follows:
+
+```
+get list of all framework monikers PS supported by the API browser
+for each NuGet package N on drop share
+    for each P in PS
+        get list of all reference assemblies AS for P
+        for each A in AS
+            get list of all APIs MS in A
+            for each M in MS
+                mark M as available for P
+```
+
+This file should be an input to the process that tags APIs with specific
+framework monikers (e.g. that tags `System.String` to be contained in `net45`,
+`net461`, `netcoreapp1.0`, `netcoreapp1.1` etc).
+
+Please note that the process as outlined above will ensure that if a package
+only provides an API for .NET Standard it will show up on all frameworks that
+implement that version of the standard, which is the desired behavior.
+Furthermore, if an API is made available for a specific framework, it will also
+show up as being available on all later versions of the same framework, which is
+also desired behavior.
```

Comment:
Hey @weshaggard. We need to think about what this means for cases where an API is moving in-box. In that case, the package needs to provide an empty folder for that framework with the special marker file `_._` -- or -- a facade that type forwards. If it's the former, we get the behavior for free. If its the latter, the indexer probably just needs to ignore type forwards, but a more correct behavior would be to follow type forwards but ignore assemblies it cannot resolve, i.e. are outside the package.

Thoughts?

Reply:
For indexing purposes I would suggest we just let whatever we define as "in the platform" override whatever might come from a nuget package for the given platform. That way we don't really have to worry about this case. There are going to be cases were we have a library in a package and one in the platform so we should just have the overriding characteristic.

Reply:
Makes sense. I've clarified this in 2e3398845d935a01fb5516bc9ecd8b6d7f92505c.